src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,146 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo,
+  isAdding: boolean,
+  isLoadingIds: number[],
+  deleteTodoFromServer: (todoId: number) => void,
+  patchTodoStatusOnServer: (todoId: number, status: boolean) => void,
+  patchTodoTitleOnServer: (todoId: number, title: string) => void,
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  isAdding,
+  isLoadingIds,
+  deleteTodoFromServer,
+  patchTodoStatusOnServer,
+  patchTodoTitleOnServer,
+}) => {
+  const { id, title, completed } = todo;
+
+  const [isDoubleClicked, setIsDoubleClicked] = useState(false);
+  const [newTitle, setNewTitle] = useState<string>(title);
+
+  const newTitleField = useRef<HTMLInputElement>(null);
+
+  const isLoading = useMemo(() => (
+    isLoadingIds.includes(id) || (isAdding && !todo.id)
+  ), [isLoadingIds, isAdding]);
+
+  const handleSubmit = useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      if (title === newTitle.trim()) {
+        setIsDoubleClicked(false);
+        setNewTitle(newTitle.trim());
+      }
+
+      if (!newTitle.trim()) {
+        deleteTodoFromServer(id);
```

Comment:
Return from function here or change logic. Now you send DELETE request and then PATCH request for the same todo.